qemu/qemu.go

```diff
@@ -26,14 +26,876 @@ package qemu
 
 import (
 	"bytes"
+	"fmt"
 	"os"
 	"os/exec"
+	"strings"
 
 	"context"
 )
 
-// LaunchQemu can be used to launch a new qemu instance by invoking the
-// qemu-system-x86_64 binary.
+// Machine describes the machine type qemu will emulate.
+type Machine struct {
+	// Type is the machine type to be used by qemu.
+	Type string
+
+	// Acceleration are the machine acceleration options to be used by qemu.
+	Acceleration string
+}
+
+// Device is the qemu device interface.
+type Device interface {
+	Valid() bool
```

Comment:
Good point. I just pushed an additional commit that adds `QemuParams()` to the Device interface and allows us to get rid of the Driver structure.
Later we will add `KernelParams()` to the device interface as well.